src/main/scala/org/constellation/rollback/RollbackLoader.scala

```diff
@@ -29,6 +30,21 @@ class RollbackLoader(
     Try(deserializeFromFile[GenesisObservation](genesisObservationPath))
       .map(Right(_))
       .getOrElse(Left(CannotLoadGenesisObservationFile(genesisObservationPath)))
+}
+
+object RollbackLoader{
+
+  private def deserializeFromFile[T](path: String): T =
+    KryoSerializer.deserializeCast[T](File(path).byteArray)
+
+  private def deserializeAllFromDirectory[T](directory: String): Seq[T] =
+    getListFilesFromDirectory(directory).map(s => deserializeFromFile[T](s))
+
+  private def getListFilesFromDirectory(directory: String): Seq[File] =
+    File(directory).list.toSeq
+
+  private def deserializeFromFile[T](file: File): T =
+    KryoSerializer.deserializeCast(file.byteArray)
```

Comment:
It should stay private in the rollback loader class.

Reply:
Should we keep them on the class if they're pure functions? Also loadSnapshotInfoSer is used in multiple places